app/models/host/discovered.rb

```
@@ -86,6 +92,9 @@ def populate_fields_from_facts facts = self.facts_hash, type = 'puppet'
       importer = super(facts)
     end
     self.subnet = Subnet.subnet_for(importer.ip)
+    atribute_set = DiscoveryAttributeSet.new()
+    atribute_set.add_discovery_attribute_set(self)
+    self.discovery_attribute_set = atribute_set if atribute_set.valid?
```

if its not valid, would it silently skip the attributes?